### ISSUE TITLE
Proposal peer review

### ISSUE THREAD

**Issue by ashleysodt14** [INFO523-Fall25-101-201/final-project-samantha-whitewater-solo#2]
The following is the peer review of the project proposal by Ashley Sodt. The team members that participated in this review are 

- Ashley Sodt - @ashleysodt14

- Describe the goal of the project.

The goal of this project is to evaluate how well different ensemble classification models predict diabetes and to also assess whether applying SMOTE improves model performance on the imbalanced dataset.

- Describe the data used or collected.

This project uses the Pima Indians Diabetes dataset, which has records for 768 adult female patients. This dataset includes eight clinical predicators and a binary diabetes outcome.

- Describe how the research question will be answered, e.g. what approaches / methods will be used.

For the first research question, model performance will be assessed by training three ensemble classifiers including Random Forest, Gradient Boosting, and AdaBoost. These classifiers will be trained on the original dataset after a train-test split and then evaluated based on accuracy, precision, recall, confusion matrices, ROC curves, and feature importance.

For the second research question, SMOTE will be applied to the training data in order to create a balanced dataset. The same ensemble models will then be retrained, and their performance on the original test set will be compared to the non-SMOTE models.

- Is there anything that is unclear from the proposal?

I found the proposal to be extremely clear, outlining concrete research questions and detailed execution plans for each. The proposal does list several evaluation tools like accuracy, precision, recall, confusion matrices, and ROC, but it does not explain how these will be used to determine which model performs best. Will the metrics be combined into a composite measure, will one metric be prioritized, or will the models be evaluated qualitatively? Some clarification here could be helpful.

- Provide constructive feedback on how the team might be able to improve their project.

After learning more about SMOTE, I understand that it creates new synthetic samples to balance the dataset. I am wondering whether additional preprocessing might be helpful to ensure SMOTE runs smoothly. For example, in the Pima dataset, I see that several clinical features contain zero values that are obviously not physiologically possible. To my understanding, SMOTE does not fix missing or invalid values, it just creates synthetic points from what it is given. So perhaps these values should be treated as missing before modeling so that SMOTE does not replicate and worsen data quality.

I am also wondering whether normalization would be necessary. As SMOTE relies on nearest-neighbor-based calculations, if features are not on the same scale, variables with large numeric ranges will potentially dominate distance measurements (ex. glucose 0-200 but DPF from 0-1) may dominate measurements. I would think that without normalization, SMOTE would look mostly at glucose similarities.

- What aspect of this project are you most interested in and would like to see highlighted in the presentation.

I am interested in seeing the overall performance of SMOTE, especially which models benefit more than others. It will also be helpful to see visual comparisons showing how model behavior changes after the data has been balanced. 

- Provide constructive feedback on any issues with file and/or code organization.

Overall, the code appears to be organized and readable to me. There are clear sections for loading packages and data as well as a clear outline of the planned modeling and evaluation workflow.

- (Optional) Any further comments or feedback?
excited to see the rest of this project :)
